### ISSUE TITLE
no optical drivers were found (Apple Silicon)

### ISSUE THREAD

**Issue by cmtkapchorowa** [13xforever/ps3-disc-dumper#68]
Of recent a small proportion of mac users have been complaining about the claim of "no optical drivers" being found on MacOS, however on the same device when a virtual machine is being utilized it works without issue.

**Comment by 13xforever** [comment]
Sounds to me they do not mount the filesystem (which happens when they pass it to VM automatically). Either way, there are no details, and I don't have a mac system, so as with all mac issues, someone will have to figure it out and submit the fix.

**Comment by bryce13950** [comment]
@cmtkapchorowa What do you mean by a virtual machine? Like within parallels, or something?

**Comment by jesusvallez** [comment]
Same problem here… How did you do it?

**Comment by ameenbadri7** [comment]
Same problem here… How to fix it ?
<img width="779" alt="Screenshot 2024-08-03 at 1 00 05 AM" src="https://github.com/user-attachments/assets/15bbd70d-d1ea-4586-9f3f-3a1c07036782">

**Comment by jlwinfre** [comment]
I ran into this issue while back-converting this to .net7 (no avalonia ui) and making a cli wrapper so that I could run it on my Intel based macbook (I am too lazy to install yet another version of dotnet and didn't want to deal with the hassle of messing with my current env/version switching; I'd rather rewrite it, so not that lazy).
Couldn't figure out why it was able to get the game info, dl/check the keys but then fail to verify the keys when enumerating drives. Knew there had to be a way to mount it to appear as an actual disc vs volume, didn't look too far since there is a free tool that does it perfectly, DaemonTools; but i will look into implementing it directly in the cli wrapper, so it will take a an iso file and do the mounting for me.
fork is [here](https://github.com/jlwinfre/ps3-disc-dumper) that runs on Intel macs and will push the cli tool shortly.

tldr;
Use DaemonTools to mount the iso's as an emulated/virtual 'Drive' and not just a mounted image.
Site: [daemon-tools.cc](https://www.daemon-tools.cc/)

**Comment by 13xforever** [comment]
@jlwinfre working with isos is not supported and I'd be _very_ cautious in your shoes.

**Comment by YoniKAr** [comment]
There is a simple fix for this instead of using apple mounting just download daemon tools and use it to mount the iso files. The disc dumper just works fine.

**Comment by sebasyw** [comment]
Try running the program through a virtual machine, it seems like the program can't detect the mounted disk through macOS built in utility.

**Comment by PortugalFromMars** [comment]
@13xforever You really like to censor people and make them look dumb. Wow such a good developer.

**Comment by joserick** [comment]
I don't know if this will help, but I had the same problem on Linux. On Ubuntu, just like on Mac, you can mount an ISO as if it were an optical drive, but PS3 Disc Dumper needs to detect a real optical drive, not just a mount point.

On Windows, this works because programs like Virtual CloneDrive or Default Mount create a virtual optical drive at the kernel level (with their own driver).

On Linux, mounting an ISO usually DOES NOT create an optical drive, which is why your tool doesn't detect it.

The solution for me was to use `cdemu`. I don't know if it also exists for Mac or if there's an alternative.

Install CDEmu
`sudo apt install gcdemu cdemu-client vhba-dkms`
Load the module
`sudo modprobe vhba`
Create a virtual optical drive
`cdemu load 0 ./path/game.iso`

And that's it, ps3-disc-dumper will detect it as an optical drive

---
**Update**: I was told that on Mac there is **"DAEMON Tools"** and it has a section called "Writable Virtual Device" that works like a "real optical drive".

**Comment by 13xforever** [comment]
No further details or reported cases were reported, closing this as invalid.
